spec/reservation_manager_spec.rb

```diff
@@ -0,0 +1,252 @@
+                                       discount_rate: 0.10)
+      expect {
+        reservation_manager.reserve(room_id: 1,
+                                    check_in_date: "2019-03-12",
+                                    check_out_date: "2019-03-14")
+      }.must_raise ArgumentError
+    end
+  end
+
+  describe "list_reservations method" do
+    it "returns an array" do
+      reservation_manager.reserve(room_id: 1,
+                                  check_in_date: "2019-03-12",
+                                  check_out_date: "2019-03-15")
+      reservation_manager.reserve(room_id: 2,
+                                  check_in_date: "2019-03-12",
+                                  check_out_date: "2019-03-15")
+      expect(reservation_manager.list_reservations(date: "2019-03-14")).must_be_kind_of Array
+    end
+
+    it "lists the right number of reservations with the given date " do
+      reservation_manager.reserve(room_id: 1,
+                                  check_in_date: "2019-03-12",
+                                  check_out_date: "2019-03-15")
+      reservation_manager.reserve(room_id: 2,
+                                  check_in_date: "2019-03-12",
+                                  check_out_date: "2019-03-15")
+      reservation_manager.reserve(room_id: 2,
+                                  check_in_date: "2019-04-12",
+                                  check_out_date: "2019-04-15")
+      expect(reservation_manager.list_reservations(date: "2019-03-14").length).must_equal 2
+      expect(reservation_manager.list_reservations(date: "2019-04-14").length).must_equal 1
+    end
+  end
+
+  describe "total_cost method" do
+    it "correctly calculates the cost for a given reservation" do
+      reservation_manager.reserve(room_id: 1, check_in_date: "2019-03-10", check_out_date: "2019-03-15")
+      expect(reservation_manager.total_cost(reservation_id: 1)).must_be_close_to 1000.0
+    end
+  end
+
+  describe "find_available_rooms" do
+    let(:reserve_03_10_03_15) { reservation_manager.reserve(room_id: 1, check_in_date: "2019-03-10", check_out_date: "2019-03-15") }
+    let(:reserve_03_17_03_20) { reservation_manager.reserve(room_id: 1, check_in_date: "2019-03-17", check_out_date: "2019-03-20") }
+    let(:reserve_04_15_04_20) { reservation_manager.reserve(room_id: 2, check_in_date: "2019-04-15", check_out_date: "2019-04-20") }
+
+    it "returns the right number of rooms given a date range" do
+      reserve_03_10_03_15
+      reserve_03_17_03_20
+      reserve_04_15_04_20
+      expect(reservation_manager.find_available_rooms(check_in_date: "2019-03-20", check_out_date: "2019-03-22").length).must_equal 20
+      expect(reservation_manager.find_available_rooms(check_in_date: "2019-04-17", check_out_date: "2019-04-19").length).must_equal 19
+      expect(reservation_manager.find_available_rooms(check_in_date: "2019-03-10", check_out_date: "2019-04-19").length).must_equal 18
```

Comment:
I would also check that at least one of the expected available rooms is available and that a room that is booked is **not** listed as available.